libtenzir/include/tenzir/plugin.hpp

```diff
@@ -726,6 +728,70 @@ class store_plugin : public virtual store_actor_plugin,
     -> std::unique_ptr<plugin_printer> final;
 };
 
+/// A printer instance class that implements the finer details of store plugin
+/// printing.
+class store_printer_impl : public printer_instance {
```

Comment:
It should be possible to leave this in the `.cpp`, I think.

Reply:
The `store_printer` needs a `store_printer_impl` definition for `return std::make_unique<store_printer_impl>(std::move(*store), ctrl);`
And as of now the `store_plugins` need the `store_printer` for `deserialize`.

Reply:
Now the `store_printer`s  are back in the `.cpp`.